### PR TITLE
Fixing build issues

### PR DESCRIPTION
**Pull request by nmittler**
Motivation:

Building against openssl and boringssl is yielding several errors. They can be supressed, but it would be better to address them.

Modifications:

Various changes to ssl.c, sslutils.c, and sslcontext.c.

Result:

Builds cleanly for both openssl and boringssl.

### CODE REVIEW

**Comment by nmittler** [comment]
@Scottmitch @normanmaurer PTAL

File: src/main/c/sslcontext.c [hunk 1]
```diff
@@ -1396,70 +1396,13 @@ TCN_IMPLEMENT_CALL(void, SSLContext, setSessionTicketKeys0)(TCN_STDARGS, jlong c
 
 /* OpenSSL end */
 
-/*
- * Adapted from Android:
- * https://android.googlesource.com/platform/external/openssl/+/master/patches/0003-jsse.patch
- */
-const char* SSL_CIPHER_authentication_method(const SSL_CIPHER* cipher){
```
Comment (normanmaurer) [inline src/main/c/sslcontext.c]: @nmittler why is this removed ? I'm a bit confused how this will still work if not boringssl is used.

Comment (nmittler) [inline src/main/c/sslcontext.c]: @normanmaurer non-boringssl should use SSL_authentication_method (which is not defined in boring).

Comment (normanmaurer) [inline src/main/c/sslcontext.c]: @nmittler are you sure "SSL_authentication_method " does the same as the method here ?

Comment (normanmaurer) [inline src/main/c/sslcontext.c]: @nmittler actually I'm very confused by this change and I think this is not correct.

Comment (nmittler) [inline src/main/c/sslcontext.c]: This method was only used by our own `SSL_authentication_method` defined just below it.  I've renamed our `SSL_authentication_method` to `authentication_method` in order to avoid the naming conflict with OpenSSL's version. I then got rid of `SSL_CIPHER_authentication_method` which also conflicts with an OpenSSL method, and we don't need anymore anyway.

File: src/main/c/sslcontext.c [hunk 2]
```diff
@@ -1124,7 +1124,7 @@ TCN_IMPLEMENT_CALL(void, SSLContext, setAlpnProtos)(TCN_STDARGS, jlong ctx, jobj
         TCN_ASSERT(ctx != 0);
         UNREFERENCED(o);
 
-        if (initProtocols(e, c->pool, &c->alpn_proto_data, &c->alpn_proto_len, alpn_protos) == 0) {
+        if (initProtocols(e, c, &c->alpn_proto_data, &c->alpn_proto_len, alpn_protos) == 0) {
```
Comment (Scottmitch) [inline src/main/c/sslcontext.c]: this is a bit scary....maybe we should make compiler warning into errors...

Luckily it looks like this parameter wasn't used in `initProtocols`...should we just remove it from the signature?

Comment (normanmaurer) [inline src/main/c/sslcontext.c]: +1 let us remove it

Comment (nmittler) [inline src/main/c/sslcontext.c]: done

File: src/main/c/ssl.c [hunk 3]
```diff
@@ -1616,7 +1617,7 @@ TCN_IMPLEMENT_CALL(jlong, SSL, getTime)(TCN_STDARGS, jlong ssl)
 
     UNREFERENCED(o);
 
-    return SSL_get_time(ssl_);
+    return SSL_get_time(ssl_->session);
```
Comment (Scottmitch) [inline src/main/c/ssl.c]: thanks @nmittler! see https://github.com/netty/netty-tcnative/issues/69.  This is used for every single `OpenSslSession` in Netty ... https://github.com/netty/netty/blob/master/handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java#L1298

**Comment by normanmaurer** [comment]
@nmittler if the netty testsuite passes with this I'm fine to cherry-pick it.

**Comment by normanmaurer** [comment]
@nmittler just ensure you squash first

**Comment by Scottmitch** [comment]
@nmittler - LGTM!

Feels like it is time to recommend that everyone update :wink:

**Comment by nmittler** [comment]
@Scottmitch @normanmaurer PTAL.  I've also cleaned up several cases of const vs non-const.

**Comment by normanmaurer** [comment]
@nmittler looks good

**Comment by nmittler** [comment]
@Scottmitch @normanmaurer I've updated the PR to include "-Werror" and got the build running cleanly on osx and ubuntu.

**Comment by normanmaurer** [comment]
@nmittler thanks! Cherry-picked as 9b06af6863bcb8c7eaf123ef7ee6134743c26690
